include/diaspora.php

```diff
@@ -2060,8 +2121,9 @@ private function original_item($guid, $orig_author, $author) {
 
 				if ($r) {
 					// If it is a reshared post from another network then reformat to avoid display problems with two share elements
-					if (self::is_reshare($r[0]["body"], false))
-						$r[0]["body"] = diaspora2bb(bb2diaspora($r[0]["body"]));
+					if (self::is_reshare($r[0]["body"], false)) {
+						$r[0]["body"] = self::replace_people_guid(diaspora2bb(bb2diaspora($r[0]["body"])), $r[0]["author-link"]);
```

Comment:
Well, it would be more an overhaul of Friendica, similar as GNUSocial has. What I mean is an Intercepting Filter-Pattern where a "filter chain" (basicly an array) is being used:

https://en.wikipedia.org/wiki/Intercepting_filter_pattern